internal/auth/providers/okta_test.go

```diff
@@ -0,0 +1,430 @@
+	s := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		rw.WriteHeader(code)
+		rw.Write(body)
+	}))
+	u, _ := url.Parse(s.URL)
+	return u, s
+}
+func newOktaProvider(providerData *ProviderData, t *testing.T) *OktaProvider {
+	if providerData == nil {
+		providerData = &ProviderData{
+			ProviderName: "",
+			ClientID:     "",
+			ClientSecret: "",
+			SignInURL:    &url.URL{},
+			RedeemURL:    &url.URL{},
+			RevokeURL:    &url.URL{},
+			ProfileURL:   &url.URL{},
+			ValidateURL:  &url.URL{},
+			Scope:        ""}
+	}
+	provider, err := NewOktaProvider(providerData, "test", "default")
+	if err != nil {
+		t.Fatalf("new okta provider returns unexpected error: %v", err)
+	}
+	return provider
+}
+
+func TestOktaProviderDefaults(t *testing.T) {
+	expectedResults := []struct {
+		name         string
+		providerData *ProviderData
+		signInURL    string
+		redeemURL    string
+		revokeURL    string
+		profileURL   string
+		userInfoURL  string
+		validateURL  string
+		scope        string
+	}{
+		{
+			name:        "defaults",
+			signInURL:   "https://test.okta.com/oauth2/default/v1/authorize",
+			redeemURL:   "https://test.okta.com/oauth2/default/v1/token",
+			revokeURL:   "https://test.okta.com/oauth2/default/v1/revoke",
+			profileURL:  "https://test.okta.com/oauth2/default/v1/userinfo",
+			validateURL: "https://test.okta.com/oauth2/default/v1/introspect",
+			scope:       "openid profile email groups offline_access",
+		},
+		{
+			name: "with provider overrides",
+			providerData: &ProviderData{
+				SignInURL: &url.URL{
+					Scheme: "https",
+					Host:   "example.com",
+					Path:   "/oauth/auth"},
+				RedeemURL: &url.URL{
+					Scheme: "https",
+					Host:   "example.com",
+					Path:   "/oauth/token"},
+				RevokeURL: &url.URL{
+					Scheme: "https",
+					Host:   "example.com",
+					Path:   "/oauth/deauth"},
+				ProfileURL: &url.URL{
+					Scheme: "https",
+					Host:   "example.com",
+					Path:   "/oauth/profile"},
+				ValidateURL: &url.URL{
+					Scheme: "https",
+					Host:   "example.com",
+					Path:   "/oauth/tokeninfo"},
+				Scope: "profile"},
+			signInURL:   "https://example.com/oauth/auth",
+			redeemURL:   "https://example.com/oauth/token",
+			revokeURL:   "https://example.com/oauth/deauth",
+			profileURL:  "https://example.com/oauth/profile",
+			userInfoURL: "https://example.com/oauth/userinfo",
+			validateURL: "https://example.com/oauth/tokeninfo",
+			scope:       "profile",
+		},
+	}
+	for _, expected := range expectedResults {
+		t.Run(expected.name, func(t *testing.T) {
+			p := newOktaProvider(expected.providerData, t)
+			if p == nil {
+				t.Errorf("Okta provider was nil")
+			}
+			if p.Data().ProviderName != "Okta" {
+				t.Errorf("expected provider name Okta, got %s", p.Data().ProviderName)
+			}
+			if p.Data().SignInURL.String() != expected.signInURL {
+				log.Printf("expected %s", expected.signInURL)
```

Comment:
instead of using the stdlib logger, we should use the test logger.

See https://golang.org/pkg/testing/#T.Logf

This makes it easier to work with limiting our log output for tests to show results where it matters.

Reply:
Good point! 👍